spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationObjectSupport.java

```diff
@@ -226,7 +226,7 @@ protected void setTaskScheduler(TaskScheduler taskScheduler) {
 	}
 
 	public ConversionService getConversionService() {
-		if (this.conversionService == null && this.beanFactory != null) {
+		if (this.conversionService == null && this.beanFactory != null) { // NOSONAR double lock fetching bean, not creating
```

Comment:
I would say we don't need this double check at all and even any synchronization here.
We really just try to obtain a bean, so even if without synchronization we would call it several times, we are going to end up with the same instance.

So, my opinion is like this:
```
if (this.conversionService == null && this.beanFactory != null) {
    this.conversionService = IntegrationUtils.getConversionService(this.beanFactory);
}
```